docs/integrations/webhooks/create-a-webhook.mdx

```diff
@@ -0,0 +1,363 @@
+
+
+  // If there are no headers, error out
+  if (!svix_id || !svix_timestamp || !svix_signature) {
+    return res.status(400).json({ error: 'Error occured -- no svix headers' })
+  }
+
+  console.log('headers', req.headers, svix_id, svix_signature, svix_timestamp)
+  // Get the body
+  const body = (await buffer(req)).toString()
+
+  // Create a new Svix instance with your secret.
+  const wh = new Webhook(WEBHOOK_SECRET);
+
+  let evt: WebhookEvent
+
+  // Verify the payload with the headers
+  try {
+    evt = wh.verify(body, {
+      "svix-id": svix_id,
+      "svix-timestamp": svix_timestamp,
+      "svix-signature": svix_signature,
+    }) as WebhookEvent
+  } catch (err) {
+    console.error('Error verifying webhook:', err);
+    return res.status(400).json({ 'Error': err })
+  }
+
+  // Do something with the payload
+  // For this guide, you simply log the payload to the console
+  const { id } = evt.data;
+  const eventType = evt.type;
+  console.log(`Webhook with and ID of ${id} and type of ${eventType}`)
+  console.log('Webhook body:', body)
+
+  return res.status(200).json({ response: 'Success' })
+}
+```
+</CodeBlockTabs>
+
+## Add your endpoint to Middleware
+
+Your Route Handler must be made public or ignored by Middleware to allow the request to succeed. The following example will make any webhooks created under `/api/webhooks/` public. See [authMiddleware](/docs/references/nextjs/auth-middleware) for more information.
+
+```tsx filename="middleware.tsx" {4}
+import { authMiddleware } from "@clerk/nextjs";
+
+export default authMiddleware({
+  publicRoutes: ["/api/webhooks(.*)"],
+});
+
+export const config = {
+  matcher: ["/((?!.+\\.[\\w]+$|_next).*)", "/", "/(api|trpc)(.*)"],
+};
+```
+
+## Test your webhook
+
+1. Start your Next.js server.
+2. On your endpoint's settings page in the Clerk Dashboard, select the **Testing** tab.
+3. In the **Select event** dropdown, select `user.updated`.
+4. Select the **Send Example** button.
+5. Below that section, in the **Message Attempts** section, you should see a successful attempt with a status of `200`.
+
+### Message failed
+
+If the message failed:
+
+1. Select the message.
+2. Scroll down to the **Webhook Attempts** section.
+3. Select the arrow icon the left side.
+4. Investigate the error. Your solution is going to depend on the error message. If you are having trouble debugging, reach out to [Clerk support](https://clerk.com/support).
+
+## Add logic for updating a user
+
+Before, you were using the Clerk Dashboard to *test* what would happen if a `user.updated` event occurred.
+
+Now, you can add logic to your application for this event to actually be able to occur.
+
+<Callout type="warning">
+  You may need to enable the **Name** setting. It can be found in the the Clerk Dashboard under User & Authentication > Email, Phone, Username > [Personal information](https://dashboard.clerk.com/last-active?path=user-authentication/email-phone-username).
```

Comment:
```suggestion
  For this example to work correctly, you may need to enable the **Name** setting. It can be found in the the Clerk Dashboard under User & Authentication > Email, Phone, Username > [Personal information](https://dashboard.clerk.com/last-active?path=user-authentication/email-phone-username).
```